app/src/main/java/com/babylon/wallet/android/presentation/common/UiMessage.kt

```diff
@@ -51,44 +51,29 @@ sealed class UiMessage(val id: String = UUIDGenerator.uuid().toString()) {
-            InvalidPayload -> context.getString(R.string.importOlympiaAccounts_invalidPayload)
-            NoMnemonicForAccounts -> context.getString(R.string.importOlympiaAccounts_noMnemonicFound)
+        @Composable
+        override fun getMessage(): String = when (this) {
+            InvalidMnemonic -> stringResource(id = R.string.importOlympiaAccounts_invalidMnemonic)
+            InvalidPayload -> stringResource(id = R.string.importOlympiaAccounts_invalidPayload)
+            NoMnemonicForAccounts -> stringResource(id = R.string.importOlympiaAccounts_noMnemonicFound)
             NoAccountsForLedger ->
                 "No addresses verified. The currently connected Ledger device is not related " +
                     "to any accounts to be imported, or has already been used."
-            is LedgerAlreadyExist -> context.getString(R.string.addLedgerDevice_alreadyAddedAlert_message, label)
-            WalletExported -> context.getString(R.string.profileBackup_manualBackups_successMessage)
-            InvalidSnapshot -> context.getString(R.string.recoverProfileBackup_incompatibleWalletDataLabel)
-            InvalidPassword -> context.getString(R.string.recoverProfileBackup_passwordWrong)
+
+            is LedgerAlreadyExist -> stringResource(id = R.string.addLedgerDevice_alreadyAddedAlert_message, label)
+            WalletExported -> stringResource(id = R.string.profileBackup_manualBackups_successMessage)
+            InvalidSnapshot -> stringResource(id = R.string.recoverProfileBackup_incompatibleWalletDataLabel)
+            InvalidPassword -> stringResource(id = R.string.recoverProfileBackup_passwordWrong)
         }
     }
 
-    @Serializable
-    @SerialName("error_message")
     data class ErrorMessage(
-        @StringRes
-        private val userFriendlyDescription: Int?,
-        private val nonUserFriendlyDescription: String?
+        private val error: Throwable?
     ) : UiMessage() {
 
-        val errorMessageStringRes: Int?
-            get() = userFriendlyDescription // TODO figure out better way how to propagate arg errors
-
-        override fun getMessage(context: Context): String = userFriendlyDescription?.let {
-            context.getString(it)
-        } ?: nonUserFriendlyDescription ?: context.getString(R.string.common_somethingWentWrong)
-
-        companion object {
-            fun from(error: Throwable?) = ErrorMessage(
-                userFriendlyDescription = (error as? DappRequestException)?.failure?.toDescriptionRes()
-                    ?: (error as? DappRequestFailure)?.toDescriptionRes(),
-                nonUserFriendlyDescription = error?.message
-            )
-        }
+        @Composable
+        override fun getMessage(): String =
+            error?.asRadixWalletException()?.toUserFriendlyMessage(LocalContext.current) ?: error?.message
+                ?: stringResource(id = R.string.common_somethingWentWrong)
```

Comment:
👌

Reply:
I would also check if error?.message is empty since there are some cases were we still return empty message exceptions in the project.